docs/tests.md

```diff
@@ -0,0 +1,439 @@
+  * Add more meaningful comments inside the test code, now it's not easy to figure out what is going on there.
+  * Fix logging here, it is quite noisy, and do flood reports.
+
+This test checks nodes are properly draining during machine deletion and this process also respects PDBs (pod disruption budgets).
+During this test - a replication controller creating with 20 desired replicas, then PDB creating for this workload.
+After the replication controller is ready, machineset marking for deletion. Along with the deletion process test checking that pods amount on the node 
+is going down to 1 and RC has at most one non-ready replica.
+
+Test considered failed in case if more than 1 replica within RC detected as not ready during machineset deletion. That would mean that PDB constraints were violated.
+
+#### Managed cluster should recover from deleted worker machines
+* Parallel: Yes
+* ExecTime: >11m on AWS
+* Recommendations:
+  * Reduce created machines number to 1 for this test, currently there are 2
+  * This test might be possibly removed since it exercises a machineset ability to recreate deleted machines and implemented with envtest
+
+This test takes machines from the machineset which was created in `BeforeEach` block and marks them for deletion,
+then waits for machines to be deleted and recreated by the machineset. Takes quite a while, because 2 machines should be deleted and recreated then.
+
+#### Managed cluster should grow and decrease when scaling different machineSets simultaneously
+* Parallel: Yes
+* ExecTime: >10m on AWS
+* Recommendations:
+  * Reduce created machines number to 1 for this test, currently there are 2
+  * Remove this test, similiar one included into origin suite. Aside of that we running this and other tests which creates and scales MSes in parallel, so we are excercising this functionality
+
+This test creates another machineset with 0 desired replicas, then it scales it up to 3 desired replicas.
+At the same time machineset which was created in `BeforeEach` block scales down to 0.
+This test considering passed when both machinesets reach desired replicas (0 for initial machineset, 3 for newly created machineset) before the timeout (30m currently).
+
+#### Managed cluster should reject invalid machinesets
+* Parallel: Yes
+* ExecTime: >6m on AWS
+* Recommendations:
+  * Get this out of this suite! No machines need to be created here.
+  * Move this to another webhook tests
+
+This test just tries to create machineset with empty provider config, but due to machineset is creating in `BeforeEach` block, it takes a while.
+We are creating and waiting for 2 machines here for nothing...
+
+#### Managed cluster should have ability to additively reconcile taints from machine to nodes
+* Parallel: Yes
+* ExecTime: >6m on AWS
+* Recommendation:
+  * Add check also for labels, or create new one which would exercise labels and annotations as well
```

Comment:
We only reconcile the labels and annotations when the node is first created, via nodelink, so I don't think this makes sense does it?